lms/services/lti_h.py

```diff
@@ -1,17 +1,39 @@
+from functools import wraps
+
 from pyramid.httpexceptions import HTTPBadRequest, HTTPInternalServerError
 
 from lms.services import HAPIError, HAPINotFoundError
 
 
+def lti_h_action(function):
+    """Disable an action if provisioning is not enabled and catch HAPIError."""
```

Comment:
This decorator simplifies the code a lot by removing the repeated disabling and catching. Doesn't have a very informative name though. `@disable_if_not_provisioned_and_catch_hapi_errors` seems a bit much

Reply:
It would still be nice to catch the exception here, as that's universal too. Although that would fix the naming issue for the decorator.

The NoOpService would work, but I wonder if we can just not call it in the first place? It's a kind of obscure way to hide the if. I can see this being one of the things that might have us scratching our heads 2 years down the line if it's still there.